WordPress/gutenberg
# Quote: Rename the align attribute to textAlign

## What?

_Updated September 27 2023._

Renames the align attribute to textAlign.
Partial for https://github.com/WordPress/gutenberg/issues/34454
There may be other blocks where the attribute should be renamed so the issue can be kept open until that is confirmed and solved.

## Why?

The block used the `align` attribute for the text alignment.
- This was inconsistent with (some) other blocks that use `align` for the block alignment and `textAlign` for text alignment.
- It conflicted with the potential addition of the align block support; see details in the issue above.

## How?

-  Renames the attribute
- Migrates the current values to the new attribute name

## Testing Instructions
_Updated September 27 2023._

1. Deactivate Gutenberg and add a quote block with centered text:

```

<blockquote class="wp-block-quote has-text-align-center">
<p>This is a quote</p>
<cite>This is a citation</cite></blockquote>

```
2. Apply the PR and activate Gutenberg.
3. Refresh the page where you added the quote. Confirm that there are no block validation errors and that the block looks the same as before.
4. Enter the code editor mode. Confirm that the code for the block uses "textAlign" instead of "align".
```

<blockquote class="wp-block-quote has-text-align-center">
<p>This is a quote</p>
<cite>This is a citation</cite></blockquote>

```

## Screenshots or screencast 
Before:
<img width="1221" alt="quote block with centered text" src="https://github.com/WordPress/gutenberg/assets/7422055/e9f78256-2df9-430a-be00-ae6fda9b0d52">

After:
<img width="1221" alt="quote block, with the pull request applied" src="https://github.com/WordPress/gutenberg/assets/7422055/af0587dc-e75d-4e8e-b993-fc5b118aebe7">
